src/SFA.DAS.LevyTransferMatching/Application/Queries/Pledges/GetApplications/GetApplicationsQueryHandler.cs

```diff
@@ -20,5 +20,82 @@ public class GetApplicationsQueryHandler : GetApplicationsQueryHandlerBase<GetAp
+        {
+            var applicationsResponse = await _levyTransferMatchingService.GetApplications(new GetApplicationsRequest
+            {
+                PledgeId = request?.PledgeId,
+                AccountId = request?.AccountId
+            });
+
+            if (applicationsResponse.Applications == null)
+            {
+                return new GetApplicationsQueryResult()
+                {
+                    Applications = null
+                };
+            }
+
+            var pledgeResponse = await _levyTransferMatchingService.GetPledge(request.PledgeId.Value);
+            var distinctStandards = applicationsResponse.Applications.Select(app => app.StandardId).Distinct();
+            var standardTasks = new List<Task<GetStandardsListItem>>(distinctStandards.Count());
+
+            standardTasks.AddRange(distinctStandards.Select(standardId => _coursesApiClient.Get<GetStandardsListItem>(new GetStandardDetailsByIdRequest(standardId))));
+
+            await Task.WhenAll(standardTasks);
+
+            foreach (var application in applicationsResponse.Applications)
+            {
+                var standard = standardTasks.Select(s => s.Result).Single(s => s.StandardUId == application.StandardId);
+                application.Standard = new Standard()
+                {
+                    LarsCode = standard.LarsCode,
+                    Level = standard.Level,
+                    StandardUId = standard.StandardUId,
+                    Title = standard.Title,
+                    Route = standard.Route,
+                    ApprenticeshipFunding = standard.ApprenticeshipFunding?.Select(funding =>
+                        new ApprenticeshipFunding()
+                        {
+                            Duration = funding.Duration,
+                            EffectiveFrom = funding.EffectiveFrom,
+                            EffectiveTo = funding.EffectiveTo,
+                            MaxEmployerLevyCap = funding.MaxEmployerLevyCap
+                        })
+                };
+
+                application.IsLocationMatch = !pledgeResponse.Locations.Any() || application.Locations.Any();
+                application.IsSectorMatch = !pledgeResponse.Sectors.Any() || application.Sectors.Any(x => pledgeResponse.Sectors.Contains(x));
+                application.IsJobRoleMatch = !pledgeResponse.JobRoles.Any() || pledgeResponse.JobRoles.Contains(application.Standard.Route);
+                application.IsLevelMatch = !pledgeResponse.Levels.Any() || pledgeResponse.Levels.Contains(MapLevelIntToString(application.Standard.Level));
+            }
+
+            return new GetApplicationsQueryResult
+            {
+                Applications = applicationsResponse.Applications.Select(x => (GetApplicationsQueryResultBase.Application)x)
+            };
+        }
+
+        public string MapLevelIntToString(int levelInt)
```

Comment:
can do this far more efficiently using substring, means that if further levels are ever added you don't have to do anything. If the penultimate value is a string then just grab the last character. If anything it's cleaner. 



Also found this (Level)request.Levels.Cast<int>() never used it but looks even better than substring, the value and the description are also the same number so potentially you can cast to string and do away with the switch statement